pandas/tests/series/test_constructors.py

```diff
@@ -2155,6 +2155,36 @@ def test_inference_on_pandas_objects(self):
             result = Series(idx)
         assert result.dtype != np.object_
 
+    def test_series_constructor_maskedarray_int_overflow(self):
+        # GH#56566
+        mx = ma.masked_array(
+            [
+                4873214862074861312,
+                4875446630161458944,
+                4824652147895424384,
+                0,
+                3526420114272476800,
+            ],
+            mask=[0, 0, 0, 1, 0],
+        )
+        result = Series(mx, dtype="Int64")
+        expected = Series(
+            IntegerArray(
+                np.array(
+                    [
+                        4873214862074861312,
+                        4875446630161458944,
+                        4824652147895424384,
+                        0,
+                        3526420114272476800,
+                    ],
+                    dtype="int64",
+                ),
+                np.array([0, 0, 0, 1, 0], dtype=np.bool_),
+            )
```

Comment:
Using the Series constructor for expected as well introduces the possibility of false "passed" test results in the future.

 I would (1) use the plain unmasked array as the expected value instead, (2) filter out the masked values for both expected and result, (3) convert the result to an array representation (np.array(Series.values) for now, since Series.to_numpy() is broken in a similar way as the constructor atm, but that's better fixed in a separate PR) and (4) assert the numpy arrays for equality.

Reply:
Thanks @dontgoto  Changed the test cases as suggested.